client/gutenberg/editor/controller.js

```diff
@@ -33,14 +33,24 @@ function getPostID( context ) {
 }
 
 export const post = ( context, next ) => {
-	const state = context.store.getState();
+	//see post-editor/controller.js for reference
 
-	const siteId = getSelectedSiteId( state );
-	const postId = getPostID( context );
 	const uniqueDraftKey = uniqueId( 'gutenberg-draft-' );
+	const postId = getPostID( context );
 	const postType = determinePostType( context );
 
-	//reserved space for adding the post to context see post-editor/controller.js
-	context.primary = <GutenbergEditor { ...{ siteId, postId, postType, uniqueDraftKey } } />;
-	next();
+	const unsubscribe = context.store.subscribe( () => {
```

Comment:
Hmm, this'll fire on every update to the redux store. Looks like we had an existing usage on the post controller. Any new alternative here @Automattic/team-calypso ?

Reply:
It'll `unsubscribe` to changes once the site state is ready, but I agree that there should be a better way to do this in the controller.

Reply:
I know that I'm very late to the review party here, but still:

1. If we want to wait until something happens in the Redux store, `subscribe` is the right way to do it. The callback will be called on every update, we should check if the update contains what we're looking for (e.g., valid selected site ID) and then unsubscribe and proceed. It's possible to write a helper that takes a callback that evaluates some condition and returns promise that resolves with some result when the condition is fulfilled. But no such thing exists in Calypso codebase AFAIK. By the way, such a `subscribe` call is what @kwight needs in #27793 🙂 

2. More importantly, I don't understand why this patch even needs to exist. The patched page.js handler waits until a valid site is selected -- but that's exactly what the `siteSelection` handler is for! And it's used for all Gutenberg routes already.

3. The page.js handlers for the Gutenberg seem a bit off:
```js
page( '/gutenberg/post', siteSelection, sites, makeLayout, clientRender );
page( '/gutenberg/post/:site/:post?', siteSelection, post, makeLayout, clientRender );
page( '/gutenberg/post/:site?', siteSelection, makeLayout, clientRender );
```

The first handler doesn't need `siteSelection` at all, because it doesn't have `:site` as route param. There's no site to select, just the `sites` (site selector UI) to render.

The second handler is OK.

The third handler is never called, as all routes it can possibly match are already matched and handled by the previous two handlers. And it's a good thing it's never called, because it would render a blank page. `siteSelection` will just set selected site in Redux, nothing else. There is no primary or secondary UI component being set, so `makeLayout+clientRender` will render an empty `<Layout>`.

I'm not sure if `/gutenberg/post/new` will work, i.e., offer site selection and then proceed to creating a new post there.